src/viur_cli/cloud.py

```diff
@@ -0,0 +1,574 @@
+                    echo_error(f"An error occurred while uploading Role data to Cloud: {e}")
+
+    except json.JSONDecodeError as e:
+        echo_fatal(f"Error decoding JSON from {roles_json_file_path}: {e}")
+    except yaml.YAMLError as e:
+        echo_fatal(f"Error opening or writing the .yaml file: {e}")
+
+
+def transform_yaml_to_dict(dict_data):
+    transformed_data = {'bindings': []}
+
+    # Create a dictionary to store unique members and their corresponding roles
+    member_roles = {}
+
+    # Iterate through the original data and organize it
+    for binding in dict_data['bindings']:
+        role = binding['role']
+        for member in binding['members']:
+            member_roles.setdefault(member, []).append(role)
+
+    # Create the transformed data structure
+    transformed_data['bindings'] = [{'members': member, 'role': roles} for member, roles in member_roles.items()]
+
+    # Add E-Tag and Version
+    transformed_data.update({"etag": dict_data["etag"], "version": dict_data["version"]})
+
+    return transformed_data
+
+
+def transform_dict_to_yaml(transformed_data):
+    original_data = {'bindings': []}
+
+    # Create a dictionary to store roles and their corresponding members
+    role_members = {}
+
+    # Iterate through the transformed data and organize it
+    for binding in transformed_data['bindings']:
+        members = binding['members']
+        roles = binding['role']
+
+        # Ensure members is a list
+        members = [members] if not isinstance(members, list) else members
+
+        for role in roles:
+            role_members.setdefault(role, []).extend(members)
+
+    # Create the original data structure
+    original_data['bindings'] = [
+        {'members': list(set(members)), 'role': role} for role, members in role_members.items()
+    ]
+
+    # Add E-Tag and Version
+    original_data.update({"etag": transformed_data["etag"], "version": transformed_data["version"]})
+
+    return original_data
+
+
+def gcloud_setup():
+    project = input("Enter PROJECT_ID: ").strip()
+
+    if not project:
+        echo_fatal("Usage: viur setup gcloud PROJECT_ID")
+        return
+
+    echo_info("Check if user is authorized with gcloud....")
+
+    try:
+        run_command("gcloud auth print-access-token")
+    except subprocess.CalledProcessError:
+        echo_warning(
+            "##############################################################\n"
+            "# Please authenticate your Google user with gcloud SDK to    #\n"
+            "# execute administrative commands.                           #\n"
+            "# In this step, a separate browser window opens to           #\n"
+            "# authenticate.                                              #\n"
+            "# This step is only required once on this computer.          #\n"
+            "##############################################################\n"
+        )
+        response = input("Are you ready?[Y/n]")
```

Comment:
Use the `click` package and its features here. This will also make gcloud_setup() parametrizable directly, so no prompts occur.